src/app/static/src/app/store/baseline/actions.ts

```diff
@@ -112,15 +112,19 @@ export const actions: ActionTree<BaselineState, DataExplorationState> & Baseline
                 .then((response) => {
                     if (response) {
                         const metadata = response.data;
-                        const pjnz = findResource(metadata, schemas.pjnz);
-                        const pop = findResource(metadata, schemas.population);
-                        const shape = findResource(metadata, schemas.shape);
-                        const survey = findResource(metadata, schemas.survey);
-                        const program = findResource(metadata, schemas.programme);
-                        const anc = findResource(metadata, schemas.anc);
+                        const avaliableResources = rootGetters ? rootGetters["baseline/selectedDatasetAvailableResources"] : {}
+                        const exceptions = { // where DatasetResource keys do not match ADRSchemas keys
+                            pop: "population",
+                            program: "programme"
+                        }
+                        const resources: { [k in keyof DatasetResourceSet]?: DatasetResource | null } = {}
+                        Object.entries(avaliableResources).forEach(([key, value]) => {
+                            const schemaKey = key in exceptions ? exceptions[key as "pop" || "program"] : key
+                            resources[key as keyof typeof resources] = value ? findResource(metadata, schemas[schemaKey as keyof ADRSchemas]) : null
+                        })
 
                         commit(BaselineMutation.UpdateDatasetResources,
-                            {pjnz, pop, shape, survey, program, anc} as DatasetResourceSet)
+                            { ...resources } as DatasetResourceSet)
```

Comment:
I haven't been able to test publishing a release manually as I don't have sufficient files to reach that stage. It looks like you were using the Imperial College Togo dataset since that doesn't have pjnz. I was using the country's dataset, which does have the pjnz file but nothing else it can use. I tried combining the files from both datasets but there were validation issues. I'll try to find a dataset without the art and/or anc files to test.

Reply:
So Senegal matches the above criteria but it's read only so although I can reach the final step I don't have the option to do a release. I don't think it's going to be possible to test this manually as any dataset where we don't have permissions for some files is going to be read only.

Reply:
OK, but looking at the logic, what does it seem to do? I guess it shouldn't give you the option to push files which aren't accessible and if that's the case for any file then it also shouldn't give you the option to do a release. I can't remember if it does any checks on this off the top of my head. Probably not? Or are we OK with them getting the option but it failing and showing an error message when user attempts to push? Probably need to check with Jeff/UNAIDS what desired behaviour is here. Jeff or Rachel might also be able to provide usable Togo pjnz to test with? Again, this could all go in another ticket, but I think we should try to get some clarity on it.